src/pint/toa.py

```diff
@@ -1277,8 +1288,12 @@ def pickle(self, filename=None):
         self.pintversion = pint.__version__
         if filename is not None:
             pickle.dump(self, open(filename, "wb"))
-        elif (self.filename is not None) and (type(self.filename) is not list):
-            pickle.dump(self, gzip.open(self.filename + ".pickle.gz", "wb"))
+        elif self.filename is not None:
+            if isinstance(self.filename, str):
+                filename = self.filename
+            else:
+                filename = self.filename[0]
```

Comment:
This is problematic, IMO, and it changes the intent of what I wrote.  In this case, if there have been merges of .tim files, the pickle file will automatically get the name of the first file.  I don't think that should happen, though.  Because that is a huge assumption that that is what you would want.  If there is more than one filename (including from an INCLUDE), I think we should have to specify the name of the pickle file.  I know that that was not the way it was before, but that's because we didn't keep track of INCLUDE filenames (which we should have).

Reply:
So as it stands, pickles are not created for merged files at all, only for files loaded from a single .tim file - which have multiple filenames now, if they have INCLUDEs. And when you have a file with INCLUDEs, the pickle gets named after the top-level file. If you are working with merged files, they have to be loaded one by one and then merged; the loading process is accelerated by pickling each of the files to be merged individually.

So your concern only applies if the user explicitly calls `.pickle` themselves.

Reply:
I'm concerned with both with merged files that a user calls pickle on (which I think they should be required to specifically give a name) and also with the pickling happening for files with INCLUDEs.  Naming after the top-level file is what we had been doing, but I'm not sure that that is what we should continue to do (there are arguments for it and also against, I think).  Personally, I think the user should be required to specify the filename if their TOAs object came from more than one file no matter how that happened.

Reply:
So I saw this in action with the `timing_analysis` from the NANOGrav Gitlab: they had to create a `.tim` file with INCLUDEs, and turning on auto-pickling made reloading `.tim` files fast. The problem was that someone might modify one of the INCLUDEd files and the cache wouldn't be invalidated - but the hashes fix that. So the pickles capture all the TOAs referred to by the top-level file, and the automatic name makes it easy to use transparent caching. 

Manual pickling is a different story; of course users can pickle any old python object to any old file and unpickle it by hand later. The point of having pickle support code here, in PINT, is to make transparent caching work. And it can, just fine, for `.tim` files with INCLUDEs. It even works for merged TOAs objects by caching their components.